### ISSUE TITLE
I have a 3D45 for you - drop me a line to jp@lihou.co.uk !  thanks

### ISSUE THREAD

**Issue by Rally037** [timmehtimmeh/Cura-Dremel-3D20-Plugin#75]

**Comment by Rally037** [comment]
Drop me a line - I have a 3D45 for you! sorry about the delay - I missed the thread expiry date.  Email is jp@lihou.co.uk

**Comment by timmehtimmeh** [comment]
@Rally037  - no worries at all.  I've sent you an email, so be on the lookout and let me know if it doesn't arrive.

FWIW you can always re-open a closed issue with the "Reopen Issue" button
![image](https://user-images.githubusercontent.com/19733103/109321104-a0a0e280-7805-11eb-8a56-b909118714a3.png)

**Comment by lenigma1** [comment]
@Rally037 Your a scholar and a gentleman!
I am humbled by your generosity to the beleaguered 3D45 community!
I'm certain im not the only one with a little buyers remorse (basically unusable with Octoprint? that's not right...), but you contribution to the gifted @timmehtimmeh will no doubt open up a few doors, for this ridiculously sheltered but very promising printer!
My deepest thank you!

**Comment by thbl** [comment]
I also have a 3D45 is there an plugin for that model ? ;D

**Comment by lenigma1** [comment]
Here.

https://github.com/Xcessity/CuraDremelSupport

For now, this is what we got to play with, and it is helpful.

Although a Dremel 3D45 is being(or has been?) supplied to timmehtimmeh,
it's reasonable to assume he's going to need some more time to unlock the
quagmire that is the 3D45 frankin-hybrid-code on our motherboards.... That
I JUST had replaced, and should be back in my hands in the coming few days!

**Comment by thbl** [comment]
@lenigma1 thanks already got that one tho ^^  jsut courious if any one had something cooler yet  ... ;D

**Comment by lenigma1** [comment]
I know Tim(for short) is hard at work with his new toy

I too am patiently waiting to see if he can crack Dremels absolutely
bazaar, and broken might I add, usage of http protocol in the file streams
being sent to the printer, that's broken ALL aftermarket software usage for
us...

Yes, technically octoprint does now work, but basically all you can do is
view printing parameters, and START a print.
No pause.
No stop.
Nadda.

Heck, I went a bought a big Tronxy 400mm corexy, had Marlin 2.0 and
octoprint up and running in the first day, including the kit thats build
from scratch, and even TronXY's broken ChiTu firmware accepts basic M
commands, and gcode input.
Alas, even my extensive knowledge with modding Pronterface couldn't get
basic commands to work on the 3D45.
Wait.
That's a lie, I remember being able to again, just start a print... But I
couldn't get the leveling sensor servo to activate to bring the sensor back
up... So had I let it continue, it would have printed, sensor down, lol.

*Good things come, to those who wait*

Cheers

**Comment by timmehtimmeh** [comment]
@lenigma1  @thbl 

No updates yet - hopefully soon.

>I too am patiently waiting to see if he can crack Dremels absolutely
bazaar, and broken might I add, usage of http protocol in the file streams
being sent to the printer, that's broken ALL aftermarket software usage for
us...

Please understand that any initial implementation of the 3D45 will be very similar to the current 3D20 implementation - it will allow the user to install the necessary printer files & material definitions via the plugin and it will generate a .g3drem file that will be exported to a USB drive.   It's likely that the first few versions will focus on tweaking the printer profile settings - after that I'll start seeing what else can be done.  Please see the discussion [here](https://github.com/timmehtimmeh/Cura-Dremel-3D20-Plugin/issues/71) on what will be initially supported and what won't be.  

Perhaps later I'll be able to do more - I've got some ideas that I'd like to explore, but before anything advanced I'm going to focus on getting everything stable & well tuned.

Cheers
Tim

**Comment by lenigma1** [comment]
I was not aware the 3D45 used, or printed with its little brothers proprietary .g3drem files!

I've always used gcode and .curamaterials(I think it's called...) And never considered it's usage in any way, lol!

The very fact you are going to focus on profile settings and tuning ~ which cannot be done very well in its stock iteration... I've got some GHASTLY pro ts to prove it ~ sends my printer soul soaring!

THANK YOU!

Although my 3D45 came back today from a 7 week stint @ the Dremel shop replacing the main board(which was faulty from day 1... My very first bed level ever failed due to z failure), I now have hope to cling to, as ABS, REAL ABS, not Dremels modified PLA "Eco-ABS", NEVER printed well, dispite dozens and dozens of settings changes and test prints!

TL;DR - I'm ecstatic with any effort being put into this expensive old girl!

I'm fully aware Octoprint functionality, if even possible, is WAY down the road.

Also, I'd be very happy to alpha/beta/control test, if needed.

**Comment by timmehtimmeh** [comment]
@lenigma1  - 👍  If / When I have an update for the 3D45 I'll post back here.  

FWIW - I was able to get octoprint working on my 3D20 - I wrote up instructions on how I did [here](https://github.com/timmehtimmeh/Cura-Dremel-3D20-Plugin/tree/stable/docs/octoprint).  Last I checked the 3D45 had some open issues on the [Octoprint plugin's github](https://github.com/Mrnt/OctoPrint-FlashForge/issues) but it looks like they were working on it.

**Comment by lenigma1** [comment]
Awesome, thanks!

**Comment by timmehtimmeh** [comment]
At long last I'm happy to announce that the new "Dremel Printer Plugin" is now in the Cura marketplace.  Huge thanks to @Rally037 and his team! 

![image](https://user-images.githubusercontent.com/19733103/125960715-c109dae2-c89e-42c5-ac2d-e3e574ee05fb.png)


I've removed the older 3D20 plugin from the marketplace, as this plugin supersedes it.  This plugin will detect if the old 3D20 plugin is installed and will prompt you to uninstall the older 3D20 plugin.  

Please note that I haven't tested the 3D40 settings for any of the Dremel materials, as I don't have access to a 3D40.  I'll continue tweaking the per-printer-per-material-per-quality settings for the Dremel materials for the 3D20 and 3D45.  I'd welcome help or recommendations on any improvements to the quality settings for the Dremel specific materials for any of the printers.  After that I'll start poking around the camera.

@lenigma1 - I'm not sure if you've seen it, but there's now a [plugin that allows you to upload to Dremel Printers](https://github.com/lwerner-lshigh/Dremel3D-CuraPrintr).  Eventually I'd like to include similar functionality in my plugin, but since there's already a plugin out there that does this I'd like to work on things that users can't do at this moment.

**Comment by lenigma1** [comment]
@timmehtimmeh I cant believe i missed your @ till now! Sorry my friend!
Yes, i did find that one, while dredging github one day, and cant wait to try it out!
BUT...
My 3D45 went to Dremel on Feb 4th for some warranty work(bad z driver), 3 months later (Dremel's official Canadian repair shop, Gilles Tool Company, ironically, is just 6 blocks north of me, ON MY STREET!), when returned by the courier, it was FAR more broken than when i sent it in.... and I'm STILL waiting for it, believe it or not!

I would be more than happy to help you out with whatever testing or quality settings you need... once i have it back!
I will keep you posted.

UPDATE: before hitting enter, i decided to check my emails once more. Apparently, Dremel Corporate has dispatched a brand new rig for me, as they where still having issues with it... another 4 months of testing/repair couldn't resolve.
Its already here in Toronto, at the repair depot, so hopefully I'll be able to use it sometime this year!

**Comment by timmehtimmeh** [comment]
@lenigma1 - sorry to hear about your issues - I'll keep my fingers crossed that your new 3D45 comes quickly!

As I said - I'm always open to any recommendations on tuning/changing the default settings for the plugin.  You can see on the README the current ["medium" quality prints](https://github.com/timmehtimmeh/Cura-Dremel-Printer-Plugin#MaterialSettings) of benchy for each of the materials - because of the magnification the pictures make the prints look worse than they do in real life, but as you can see there's definitely room for improvement.  When you get your printer back, and have a chance to test the plugin please feel free to open a new issue with any recommended changes (or make a pull request if you're so inclined) - I'll then go test the changes on my 3D45 and we can start a discussion then.

FWIW I'm about to push out version 0.7.1 of the plugin that tweaks the acceleration values for the printers - see https://github.com/timmehtimmeh/Cura-Dremel-Printer-Plugin/issues/81 for the discussion there if you're interested.

Cheers
Tim
